src/test/java/seedu/address/model/food/MacronutrientTest.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.food;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class MacronutrientTest {
+
+    private static class MacronutrientObjectCreator extends Macronutrient {
+        MacronutrientObjectCreator(String name, int amount, int caloriMultiplier) {
+            super(name, amount, caloriMultiplier);
+        }
+    }
+
+    private static final Fat DEFAULT_FAT_1 = new Fat(1);
+    private static final Fat DEFAULT_FAT_2 = new Fat(1);
+    private static final Fat DEFAULT_FAT_3 = new Fat(2);
+
+
+    private static final Protein DEFAULT_PROTEIN_1 = new Protein(1);
+
+    private static final Macronutrient MACRONUTRIENT_1 = new MacronutrientObjectCreator("dummy", 4, 9);
+    private static final Macronutrient MACRONUTRIENT_2 = new MacronutrientObjectCreator("dummy", 9, 4);
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () ->
+            new MacronutrientObjectCreator(null, 1, 1));
+    }
+
+    @Test
+    public void amount_lesserThanZero_throwIllegalArgumentException() {
+
```

Comment:
Should remove this or put a TODO comment if its blank

Reply:
Oops, I forgot to implement it. I will implement it in the next commits